servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/RetryingHttpRequesterFilter.java

```diff
@@ -0,0 +1,693 @@
+                if (exponential) {
+                    assert maxDelay != null;
+                    return jitter == FULL_JITTER ?
+                            retryWithExponentialBackoffFullJitter(
+                                    maxRetries, t -> true, initialDelay, maxDelay, effectiveExecutor) :
+                            retryWithExponentialBackoffDeltaJitter(
+                                    maxRetries, t -> true, initialDelay, jitter, maxDelay, effectiveExecutor);
+                } else {
+                    return jitter == FULL_JITTER ?
+                            retryWithConstantBackoffFullJitter(
+                                    maxRetries, t -> true, initialDelay, effectiveExecutor) :
+                            retryWithConstantBackoffDeltaJitter(
+                                    maxRetries, t -> true, initialDelay, jitter, effectiveExecutor);
+                }
+            }
+        }
+    }
+
+    /**
+     * An interface that enhances any {@link Exception} to provide a constant {@link Duration delay} to be applied when
+     * retrying through a {@link RetryingHttpRequesterFilter retrying-filter}.
+     * <p>
+     * Constant delay returned from {@link #delay()} will be additive to the backoff policy defined for a certain
+     * retry-able failure.
+     */
+    public interface DelayedRetry {
+
+        /**
+         * A constant delay to apply in milliseconds.
+         * The total delay for the retry logic will be the sum of this value and the result of the
+         * {@link BackOffPolicy back-off policy} in-use. Consider using 'full-jitter'
+         * flavours from the {@link BackOffPolicy} to avoid having another constant delay applied per-retry.
+         *
+         * @return The {@link Duration} to apply as constant delay when retrying.
+         */
+        Duration delay();
+    }
+
+    /**
+     * A builder for {@link RetryingHttpRequesterFilter}, which puts an upper bound on retry attempts.
+     * To configure the maximum number of retry attempts see {@link #maxTotalRetries(int)}.
+     */
+    public static final class Builder {
+        private boolean waitForLb = true;
+        private boolean ignoreSdErrors;
+
+        private int maxRetries = 3;
+
+        @Nullable
+        private Function<HttpResponseMetaData, HttpResponseException> responseMapper;
+
+        private BiFunction<HttpRequestMetaData, RetryableException, BackOffPolicy>
+                retryRetryableExceptions = (requestMetaData, e) -> ofInstant();
+
+        @Nullable
+        private BiFunction<HttpRequestMetaData, IOException, BackOffPolicy>
+                retryIdempotentRequests;
+
+        @Nullable
+        private BiFunction<HttpRequestMetaData, DelayedRetry, BackOffPolicy>
+                retryDelayedRetries;
+
+        @Nullable
+        private BiFunction<HttpRequestMetaData, Throwable, BackOffPolicy>
+                retryOther;
```

Comment:
How do you feel, should we have another `BiFunction` for `HttpResponseException` or we expect users to handle it by `retryOther`?

Reply:
I was 50-50 about that, since you mentioned it, I will add.